apps/web/components/v2/settings/webhook/WebhookForm.tsx

```diff
@@ -0,0 +1,265 @@
+import { useState, useEffect } from "react";
+import { useForm, Controller } from "react-hook-form";
+
+import { classNames } from "@calcom/lib";
+import { WEBAPP_URL } from "@calcom/lib/constants";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import { WebhookTriggerEvents } from "@calcom/prisma/client";
+import { inferQueryOutput, trpc } from "@calcom/trpc/react";
+import Button from "@calcom/ui/v2/core/Button";
+import Switch from "@calcom/ui/v2/core/Switch";
+import Select from "@calcom/ui/v2/core/form/Select";
+import { Form, Label, Input, TextField, TextArea } from "@calcom/ui/v2/core/form/fields";
+
+import { WEBHOOK_TRIGGER_EVENTS_GROUPED_BY_APP_V2 } from "@lib/webhooks/constants";
+import customTemplate, { hasTemplateIntegration } from "@lib/webhooks/integrationTemplate";
+
+import WebhookTestDisclosure from "@components/v2/settings/webhook/WebhookTestDisclosure";
+
+export type TWebhook = inferQueryOutput<"viewer.webhook.list">[number];
+
+export type WebhookFormData = {
+  id?: string;
+  subscriberUrl: string;
+  active: boolean;
+  eventTriggers: WebhookTriggerEvents[];
+  secret: string | null;
+  payloadTemplate: string | undefined | null;
+};
+
+export type WebhookFormSubmitData = WebhookFormData & {
+  changeSecret: boolean;
+  newSecret: string;
```

Comment:
These are not a part of the webhook data but used to check if the webhook secret has changed.